controller/control.go

```diff
@@ -1035,3 +1035,14 @@ func (c *Controller) monitoring(address string, backend types.Backend) {
 	logrus.Infof("Monitoring stopped %v", address)
 	c.RemoveReplicaNoLock(address)
 }
+
+func (c *Controller) monitorPreload(address string) {
+	logrus.Infof("Get preload status of backend %s", address)
+	if err := c.backend.GetPreloadStatus(address); err != nil {
+		logrus.Errorf("Failed to monitor preload, error: %v", err)
+		return
+	}
+	c.startFrontend()
+	c.setReplicaModeNoLock(address, types.RW)
```

Comment:
why 'NoLock'?

Reply:
There is no specific reason, since setReplicaModeNoLock() was getting used that's why used the same, because logic is no different that earlier.

Reply:
its not about that current code. Won't we be needing lock here?

Reply:
We are entering by taking lock, that's why we donot need another lock.